chia/util/keychain.py

```diff
@@ -87,11 +87,32 @@ def bytes_to_mnemonic(mnemonic_bytes: bytes) -> str:
     return " ".join(mnemonics)
 
 
-def bytes_from_mnemonic(mnemonic_str: str) -> bytes:
+def mnemonic_from_short_words(mnemonic_str: str) -> str:
+    """
+    Since the first 4 letters of each word is unique (or the full word, if less than 4 characters), and its common
+    practice to only store the first 4 letters of each word in many offline storage solutions, also support looking
+    up words by the first 4 characters
+    """
     mnemonic: List[str] = mnemonic_str.split(" ")
     if len(mnemonic) not in [12, 15, 18, 21, 24]:
         raise ValueError("Invalid mnemonic length")
 
+    four_char_dict = {word[: min(4, len(word))]: word for word in bip39_word_list().splitlines()}
```

Comment:
Slicing already does this for you.  I forget, black may reformat this.

```suggestion
    four_char_dict = {word[:4]: word for word in bip39_word_list().splitlines()}
```